### DISCUSSION TITLE
Rollup build with Circular Dependencies Warnings

### DISCUSSION THREAD

**Discussion by gabrsar** [directus/directus#15229] [Archive]
Hi! I'm getting some warnings about circular dependencies while building my extensions.

After some googling, I found that the cause might be the Rollup building process.

Example:
```
Circular dependency: ../../../node_modules/@redis/json/dist/commands/index.js -> ../../../node_modules/@redis/json/dist/commands/ARRINDEX.js -> ../../../node_modules/@redis/json/dist/commands/index.js
Circular dependency: ../../../node_modules/luxon/src/datetime.js -> ../../../node_modules/luxon/src/duration.js -> ../../../node_modules/luxon/src/impl/locale.js -> ../../../node_modules/luxon/src/datetime.js
Circular dependency: ../../../node_modules/joi/lib/common.js -> ../../../node_modules/joi/lib/messages.js -> ../../../node_modules/joi/lib/template.js -> ../../../node_modules/joi/lib/common.js

```
And so on.

[Some places fix it by configuring Rollup to just ignore them](https://github.com/rollup/rollup/issues/1089#issuecomment-402109607) as they look "harmless". Others [uses a Rollup plugin called "typescript" so it](https://github.com/rollup/rollup/issues/1089#issuecomment-402109607) can better understand what is happening in the code and don't mess up with dependencies and so on.

I would like to know if someone ever had this kind of issue, and how to deal with it, as I couldn't find how to access the `rollup.config.js` file and the `typescript-plugin` didn't worked to me.

I event read the [directus docs about rollup](https://docs.directus.io/extensions/creating-extensions.html#building-your-extension), but its quite vague about what I can do and what I can't. 

Thank you.

**Comment by br41nslug** [comment]
I've ran into this before with certain dependencies but have generally ignored it or opted for another library if it really didnt work 😬 
> as I couldn't find how to access the rollup.config.js file and the typescript-plugin didn't worked to me.

The rollup config is hidden away if you're using the directus extension-sdk, unfortunately the only way to edit this is to copy the used configs from the package and start managing the rollup config manually.

**Comment by rijkvanzanten** [comment]
This looks very similar to https://github.com/directus/directus/discussions/14366

Node's CJS -> ESM transition has caused massive headaches before.. We're actually biting the bullet and moving everything over to ESM across the board to hopefully prevent some of these issues moving forwards (#15190)
